src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs

```diff
@@ -88,11 +108,21 @@ private int SignPackage(string packagePath, ILogger logger, ISignatureProvider s
                 signer.SignAsync(request, logger, CancellationToken.None).Wait();
             }
 
-            OverwritePackage(tempFilePath, packagePath);
-            
+            OverwritePackage(tempFilePath, outputPath);
+
+            CleanUp(tempFilePath);
```

Comment:
Use this instead, it is more robust

https://github.com/NuGet/NuGet.Client/blob/9d1459f473c007f04b7635c0a75fd34ed08b29f3/src/NuGet.Core/NuGet.Common/PathUtil/FileUtility.cs#L198